src/mode/markdown_highlight_rules.js

```diff
@@ -3,183 +3,1165 @@
 var modes = require("../config").$modes;
 
 var oop = require("../lib/oop");
-var lang = require("../lib/lang");
 var TextHighlightRules = require("./text_highlight_rules").TextHighlightRules;
-var HtmlHighlightRules = require("./html_highlight_rules").HtmlHighlightRules;
-
-var escaped = function(ch) {
-    return "(?:[^" + lang.escapeRegExp(ch) + "\\\\]|\\\\.)*";
-};
+var lang = require("../lib/lang");
 
-var MarkdownHighlightRules = function() {
-    HtmlHighlightRules.call(this);
+var tagMap = lang.createMap({
+    a: 'anchor',
+    button: 'form',
+    form: 'form',
+    img: 'image',
+    input: 'form',
+    label: 'form',
+    option: 'form',
+    script: 'script',
+    select: 'form',
+    textarea: 'form',
+    style: 'style',
+    table: 'table',
+    tbody: 'table',
+    td: 'table',
+    tfoot: 'table',
+    th: 'table',
+    tr: 'table'
+});
+var MarkdownHighlightRules = function () {
     // regexp must not have capturing parentheses
     // regexps are ordered -> the first match is used
     var codeBlockStartRule = {
-        token : "support.function",
-        regex : /^\s*(```+[^`]*|~~~+[^~]*)$/,
-        onMatch: function(value, state, stack, line) {
-            var m = value.match(/^(\s*)([`~]+)(.*)/);
+        token: "support.function",
+        regex: /\s*(```+[^`]*|~~~+[^~]*)/,
+        onMatch: function (value, scope, stack, line) {
+            var m = line.match(/^(\s*)([`~]+)(.*)/);
             var language = /[\w-]+|$/.exec(m[3])[0];
             // TODO lazy-load modes
-            if (!modes[language])
-                language = "";
-            stack.unshift("githubblock", [], [m[1], m[2], language], state);
-            return this.token;
+            if (!modes[language]) language = "";
+            if (scope.name == "blockquote") {
+                return scope.parent.get("start").get(this.token);
+            }
+            
+            var parent = scope.get("codeBlock" + language);
+            parent.language = language;
+            parent.indent = m[1].length;
+            parent.endMarker = m[2];
```

Comment:
this code is broken because it is changing properties on the same scope object, it should be something like parent.get(this.next, {language, indent, marker, toString() { [language, indent, marker].join("|") }})`